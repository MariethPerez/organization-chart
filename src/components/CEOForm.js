import React from "react";
import styled from '@emotion/styled'



function CEOForm() {
  return (
  <h1>CEOForm</h1>)
};

export default CEOForm;
