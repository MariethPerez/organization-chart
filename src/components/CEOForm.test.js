import React from "react";
import { render, fireEvent } from "@testing-library/react";

import CEOForm from "./CEOForm";

test("CEOForm", () => {
  const { asFragment } = render(<CEOForm />);

  expect(asFragment()).toMatchSnapshot();
});

test("CEOForm functionality", () => {
  const setCeoName = jest.fn();
  const { asFragment, getByLabelText } = render(
    <CEOForm setCeoName={setCeoName} />
  );

  expect(asFragment()).toMatchSnapshot();
  const input = getByLabelText("Input to get CEO's name");

  fireEvent.change(input, {
    target: { value: "CESAR" }
  });

  fireEvent.submit(input);

  expect(setCeoName).toHaveBeenCalledWith("CESAR");
});
