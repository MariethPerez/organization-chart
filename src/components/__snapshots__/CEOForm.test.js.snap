// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CEOForm 1`] = `
<DocumentFragment>
  .emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100vh;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 500px;
  padding: 1.5em;
  background: black;
  border-radius: 0.5em;
}

.emotion-0 {
  text-align: center;
  margin: 15px;
  font-size: 40px;
  color: white;
  font-weight: bolder;
}

.emotion-1 {
  margin: 1em 0;
  font-size: 30px;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 4px;
  outline: none;
  border: none;
  border-bottom: 1px solid #ddd;
  width: 100%;
  text-align: center;
}

.emotion-1:focus {
  border: 1px solid red;
}

.emotion-2 {
  line-height: 1.5em;
  background: darkgray;
  border: none;
  color: white;
  font-size: 20px;
  font-weight: bold;
  padding: 0.5em;
  cursor: pointer;
  border-radius: 0.25rem;
  text-align: center;
  width: 50%;
}

.emotion-2:hover {
  background-color: #484848;
}

<div
    class="emotion-4"
  >
    <form
      class="emotion-3"
    >
      <h1
        class="emotion-0"
      >
        CEO's name
      </h1>
      <input
        aria-label="Input to get CEO's name"
        autocomplete="off"
        class="emotion-1"
        id="ceo"
        name="ceo"
        placeholder="Enter CEO's name"
        required=""
        type="text"
      />
      <button
        class="emotion-2"
        type="submit"
      >
        Next
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`CEOForm functionality 1`] = `
<DocumentFragment>
  .emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100vh;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 500px;
  padding: 1.5em;
  background: black;
  border-radius: 0.5em;
}

.emotion-0 {
  text-align: center;
  margin: 15px;
  font-size: 40px;
  color: white;
  font-weight: bolder;
}

.emotion-1 {
  margin: 1em 0;
  font-size: 30px;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 4px;
  outline: none;
  border: none;
  border-bottom: 1px solid #ddd;
  width: 100%;
  text-align: center;
}

.emotion-1:focus {
  border: 1px solid red;
}

.emotion-2 {
  line-height: 1.5em;
  background: darkgray;
  border: none;
  color: white;
  font-size: 20px;
  font-weight: bold;
  padding: 0.5em;
  cursor: pointer;
  border-radius: 0.25rem;
  text-align: center;
  width: 50%;
}

.emotion-2:hover {
  background-color: #484848;
}

<div
    class="emotion-4"
  >
    <form
      class="emotion-3"
    >
      <h1
        class="emotion-0"
      >
        CEO's name
      </h1>
      <input
        aria-label="Input to get CEO's name"
        autocomplete="off"
        class="emotion-1"
        id="ceo"
        name="ceo"
        placeholder="Enter CEO's name"
        required=""
        type="text"
      />
      <button
        class="emotion-2"
        type="submit"
      >
        Next
      </button>
    </form>
  </div>
</DocumentFragment>
`;
